pkgs/development/libraries/science/math/petsc/default.nix

```diff
@@ -1,4 +1,20 @@
-{ lib, stdenv , darwin , fetchurl , blas , gfortran , lapack , python }:
+{ lib
+, stdenv
+, fetchurl
+, darwin
+, gfortran
+, python
```

Comment:
Can this be python3? Python2 should no longer be used in new things.

Reply:
Yeah, we left it that way because the scripts work in both Python2 and Python3, and some ancient HPC systems with large user bases still have Python2. `python` is Python3 on some systems (since ~2011 on Arch).

Reply:
What is preferred then, hardcoding `python3` here or leaving it at `python`, which can be overridden system wide as needed (myself favoring the latter, but I'm no python expert).

Reply:
Python2 is EOL and should no longer be used in new packages. Just use python3.

> leaving it at python, which can be overridden system wide as needed (myself favoring the latter, but I'm no python expert).

I am not sure if this is something nixpkgs officially supports.